Framework/Geometry/inc/MantidGeometry/Rendering/ShapeInfo.h

```diff
@@ -0,0 +1,91 @@
+#ifndef MANTID_GEOMETRY_SHAPEINFO_H_
+#define MANTID_GEOMETRY_SHAPEINFO_H_
+
+#include "MantidGeometry/DllConfig.h"
+#include <vector>
+
+namespace Mantid {
+namespace Kernel {
+class V3D;
+}
+namespace Geometry {
+class RectangularDetector;
+class StructuredDetector;
+class IObjComponent;
+
+/** ShapeInfo : Stored Object related shape information for rendering.
```

Comment:
* Minor point. Description could be made clearer.
* This class looks like it has a creational/building job for other objects an attributes rather than `info` status? 
* Has it been checked that it is not possible to get into a bad state with this type? What happens if you call different setters on the same `ShapeInfo` object?

Reply:
* All members are overwritten with every setter so calling the setters multiple times should not result in invalid configurations, in the absence of concrete types for Cones, Spheres etc, there is no point forcing this in the constructors.
* I did think of `ShapeCreator` but although this object seems to create shapes, it actually stores information about the shape of an object which can be retrieved at the point of rendering. I cannot think of a better name.
*